rest/handler.go

```diff
@@ -209,17 +213,24 @@ func (h *handler) invoke(method handlerMethod, accessPermissions []Permission, r
 		if dbContext, err = h.server.GetDatabase(dbname); err != nil {
 			base.Infof(base.KeyHTTP, "Error trying to get db %s: %v", base.MD(dbname), err)
 
-			if shouldCheckAdminAuth {
-				if httpError, ok := err.(*base.HTTPError); ok && httpError.Status == http.StatusNotFound {
+			if httpError, ok := err.(*base.HTTPError); ok && httpError.Status == http.StatusNotFound {
+				if shouldCheckAdminAuth {
 					authorized, err := h.checkAdminAuthenticationOnly()
 					if err != nil {
 						return err
 					}
 
-					if authorized {
-						return base.HTTPErrorf(http.StatusForbidden, "")
+					if !authorized || h.providedAuthCredentials() {
```

Comment:
On master we only use providedAuthCredentials on the public port case - it looks like we're using it in conjunction with admin auth here (inside shouldCheckAdminAuth), which seems incorrect?  Which combination is this handling?

Reply:
Seems to be used.